### ISSUE TITLE
Error on model.summary() with PaliGemma

### ISSUE THREAD

**Issue by bebechien** [keras-team/keras-hub#1654]
**Describe the bug**

model.summary() for PaliGemma will fail when passing `show_trainable=True`\
Works fine without the parameter.\
Below is an actual error message.

```
Preprocessor: "pali_gemma_causal_lm_preprocessor"
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ Tokenizer (type)                                   ┃                                             Vocab # ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ pali_gemma_tokenizer (PaliGemmaTokenizer)          │                                             257,152 │
└────────────────────────────────────────────────────┴─────────────────────────────────────────────────────┘
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
[<ipython-input-2-0216a872cf53>](https://localhost:8080/#) in <cell line: 1>()
----> 1 gemma_lm.summary(show_trainable=True)

2 frames
[/usr/local/lib/python3.10/dist-packages/keras/src/utils/summary_utils.py](https://localhost:8080/#) in get_layer_fields(layer, prefix)
    262             fields.append(get_connections(layer))
    263         if show_trainable:
--> 264             if layer.weights:
    265                 fields.append(
    266                     bold_text("Y", color=34)

AttributeError: 'GetItem' object has no attribute 'weights'
```

**To Reproduce**
https://colab.research.google.com/drive/1o04v1Xok9892KC-TUOuYh8ToU2wY8RPY?usp=sharing

**Expected behavior**
Display the model summary with Trainable info.

**Additional context**
N/A

**Would you like to help us fix it?**
Sure

**Comment by miticollo** [comment]
This is a semantic issue.

`AttributeError` is raised [here](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/utils/summary_utils.py#L264) because a `keras.ops` (like [`GetItem`](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/ops/numpy.py#L2568-L2638)) doesn't have the [`weights`](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/layers/layer.py#L599-L618) property.
In Keras, a `Layer` [is](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/layers/layer.py#L60) an `Operation`, but vice versa is not true. Anyway, a `Functional` model can [also include](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/models/functional.py#L49-L59) raw Keras 3 ops.
Maybe, a dirty workaround could be to add the `weights` property to `Operation` class:
```python
@property
def weights(self):
    weights = []
    return weights
```
IMO this solution is not semantically correct because weights for an operation don't make any sense.

Or an alternative solution could be to wrap the `keras.ops` into a Lambda layer.I prefer avoiding this layer because of [its limitations](https://github.com/keras-team/keras/blob/11be99a6b4ea050528ba1739cad8bf969e242d4c/keras/src/layers/core/lambda_layer.py#L22-L31) but maybe in this case it can be used safe.

Finally this issue should be moved in [`keras`](https://github.com/keras-team/keras) repository.

**Comment by mattdangerw** [comment]
Thanks @miticollo. Agreed, this should move to the `keras` repo.

Could we just have `models.layers` filter down to only return `Layer` and not `Operation`? That's what is used by the summary utils, and generally speaking I think having `model.layers` return something that is not a `Layer` is just unexpected.

Not sure if anything would break, but we could try it out. @miticollo are you interested in taking it on?

**Comment by miticollo** [comment]
Thank u for the answer! At the moment I'm a bit busy with the latest exams at University (one is on Deep Learning). So I can't do it.
Anyway using a filter it could be a good a idea. But currently `summary` shows also the `keras.ops`. So is this behaviour still intended?

**Comment by miticollo** [comment]
Hi @mattdangerw!

This issue can be closed. It was solved with [this commit](https://github.com/keras-team/keras/commit/933579d3c4a585f236982d05d3a74921f9567415) and it included into Keras 3.5.0.

**Comment by Gopi-Uppari** [comment]
Hi @bebechien,

Could you please confirm if this issue is resolved for you with the above comment ? Please feel free to close the issue if it is resolved ?

Thank you.
